src/main/java/org/apache/mesos/specification/DefaultPlanSpecificationFactory.java

```diff
@@ -0,0 +1,21 @@
+package org.apache.mesos.specification;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class provides a default implementation of the PlanSpecifiactionFactory interface.
+ */
+public class DefaultPlanSpecificationFactory implements PlanSpecificationFactory {
+    @Override
+    public PlanSpecification getPlanSpecification(ServiceSpecification serviceSpecification) {
+        List<PhaseSpecification> phaseSpecifications = new ArrayList<>();
+
+        for (TaskTypeSpecification taskTypeSpecification : serviceSpecification.getTaskSpecifications()) {
+            phaseSpecifications.add(
+                    new DefaultPhaseSpecification(taskTypeSpecification.getName(), taskTypeSpecification));
```

Comment:
See note above -- DefaultPhaseSpecification could accept just the taskTypeSpecification and call getName() internally.